### ISSUE TITLE
`*` doesn't work in text mode

### ISSUE THREAD

**Issue by valrus** [syl20bnr/spacemacs#380]
Repro:

Open a text mode buffer, insert text
```
blah
blah
```
Put point on the first `blah` and hit `*`

Result (traceback):
```elisp
Debugger entered--Lisp error: (wrong-type-argument stringp none)
  propertize(none face ahs-plugin-defalt-face)
  ahs-decorated-current-plugin-name()
  ahs-stat()
  (let* ((i 0) (overlay-count (length ahs-overlay-list)) (overlay (format "%s" (nth i ahs-overlay-list))) (current-overlay (format "%s" ahs-current-overlay)) (st (ahs-stat)) (plighter (ahs-current-plugin-prop (quote lighter))) (plugin (format " <%s> " (cond ((string= plighter "HS") "D") ((string= plighter "HSA") "B") ((string= plighter "HSD") "F")))) (propplugin (propertize plugin (quote face) (list (quote :foreground) "#ffffff" (quote :background) (face-attribute (quote ahs-plugin-defalt-face) :foreground))))) (while (not (string= overlay current-overlay)) (setq i (1+ i)) (setq overlay (format "%s" (nth i ahs-overlay-list)))) (let* ((x/y (format "[%s/%s]" (- overlay-count i) overlay-count)) (propx/y (propertize x/y (quote face) ahs-plugin-whole-buffer-face)) (hidden (if (< 0 (- overlay-count (nth 4 st))) "*" "")) (prophidden (propertize hidden (quote face) (quote (:weight bold))))) (echo "%s %s%s press (n/N) to navigate, (e) to edit, (r) to change range or (R) for reset" propplugin propx/y prophidden)))
  spacemacs/auto-highlight-symbol-overlay-map()
  ad-Advice-ahs-forward(#[nil "\300\301\302\"\207" [ahs-select ahs-forward-p t] 3 ("c:/Users/valrus/AppData/Roaming/.emacs.d/elpa/auto-highlight-symbol-20130313.243/auto-highlight-symbol.elc" . 36082) nil])
  apply(ad-Advice-ahs-forward #[nil "\300\301\302\"\207" [ahs-select ahs-forward-p t] 3 ("c:/Users/valrus/AppData/Roaming/.emacs.d/elpa/auto-highlight-symbol-20130313.243/auto-highlight-symbol.elc" . 36082) nil] nil)
  ahs-forward()
  (progn (spacemacs/integrate-evil-search t) (ahs-highlight-now) (ahs-forward))
  eval((progn (spacemacs/integrate-evil-search t) (ahs-highlight-now) (ahs-forward)) nil)
  spacemacs/quick-ahs-forward()
  call-interactively(spacemacs/quick-ahs-forward nil nil)
  command-execute(spacemacs/quick-ahs-forward)
```

I opened https://github.com/gennad/auto-highlight-symbol/issues/2 for this as well but I don't know that I have any hope of it ever being addressed. We also discussed this in https://github.com/syl20bnr/spacemacs/issues/133 where I erroneously thought it was because of non-ASCII text, but it fails even in a text buffer that's all ASCII.

**Comment by syl20bnr** [comment]
Thank you for the report, in the worst case I'll fork it.

**Comment by herbertjones** [comment]
The error comes from certain variables not having been intialized for the local buffer.  This happens when the ahs minor mode isn't enabled for that mode.  Enabling the minor mode by calling `auto-highlight-symbol-mode` will prevent the errors.  I'm not sure yet how to enable `auto-highlight-symbol-mode` for all modes, though I imagine it may be this way for certain modes that are initialized before `(global-auto-highlight-symbol-mode t)` is called.

**Comment by syl20bnr** [comment]
Thank you @herbertjones 
Might be a duplicate of https://github.com/syl20bnr/spacemacs/blob/master/spacemacs/packages.el#L2137

**Comment by syl20bnr** [comment]
Fixed by c21bd2d2cf965e32be3eac4239b3b97795969e61
